### ISSUE TITLE
Non Global Installation Method

### ISSUE THREAD

**Issue by danielhelfand** [Azure/azure-cli#6901]
---

### Environment summary

Install Method 
No available method 

CLI version 
All 2.0 versions

OS version
CentOS 7

I would like to see an installation method available for the 
Azure CLI that allows you to do two things:

1. Download specific versions of the CLI
2. Download to a specific directory (not installed globally)

You can currently do this with pip as referenced in #5223, but I cannot use pip as it causes python path issues for my use case. I think being able to run a curl command to get installation files followed by being able to run an installer to set it up in a directory would be great.

**Comment by yugangw-msft** [comment]
One option is to have a tarball.gz file which has Python included like we did for Linux bundles and all laid out in a right way so the "az.sh" can find it.

**Comment by danielhelfand** [comment]
@yugangw-msft I agree with the bundle approach. `tarball.gz` would be a workable format as well.

**Comment by yonzhan** [comment]
issue triaged.
